addons/hr/tests/test_hr_contract_versions.py

```diff
@@ -0,0 +1,652 @@
+            'country_id': cls.env.ref('base.us').id,
+        })
+        cls.env.user.company_id = cls.company
+        cls.employee = cls.env['hr.employee'].create({
+            'name': 'John Doe',
+            'date_version': '2025-01-01'
+        })
+
+    def create_contract_with_versions(self, date_start, date_end, base_version, other_versions_to_create=[]):
+        base_version.update({
+            'contract_date_start': date_start,
+            'contract_date_end': date_end,
+        })
+        other_versions_in_contract = self.env['hr.version']
+        for date_version in other_versions_to_create:
+            version = self.employee.create_version({'date_version': date_version})
+            if version.contract_date_start:
+                other_versions_in_contract |= version
+        return base_version | other_versions_in_contract
+
+    def assert_contract_versions(self, date_start, date_end, number_of_contract_expected, versions_per_contract_expected):
+        versions_per_contract = self.employee._get_contract_versions(date_start, date_end)[self.employee.id]
+        self.assertEqual(len(versions_per_contract), number_of_contract_expected, "%s contract should be found" % number_of_contract_expected)
+        for vpc, vpc_e in zip(versions_per_contract.values(), versions_per_contract_expected):
+            self.assertEqual(vpc, vpc_e, "invalid number of versions (%s instead of %s) for this contract : contract_date_start : %s" % (len(vpc), len(vpc_e), vpc_e[0].contract_date_start))
+
+    def assert_contract(self, date_start, date_end, use_latest_version, contract_expected):
+        contracts = self.employee._get_contracts(date_start, date_end, use_latest_version)[self.employee.id]
+        for c, c_e in zip(contracts, contract_expected):
+            self.assertEqual(c, c_e, "invalid contracts")
+
+    """ Timeline for this setup and tests
+        V  : versions
+        C  : first version of the contract
+        =  : contract
+
+        1/
+                     04/01               7/31
+        2025|C---------=====================----------|
+           01/01               06/01                12/01
+
+        2/
+                     04/01               7/31
+        2025|----------==========C==========----------|
+           01/01               06/01                12/01
+
+        3/
+                     04/01               7/31
+        2025|----------=====================---------C|
+           01/01               06/01                12/01
+    """
+    def setup_1contract_1version(self):
+        expected_versions_contract = self.create_contract_with_versions(
+            "2025-04-01",
+            "2025-07-31",
+            self.employee.version_id,
+        )
+        return dict(
+            expected_versions_contract=expected_versions_contract
+        )
+
+    def test_1contract_1version(self):
+        """
+        We should retrieve the contract
+        """
+        res = self.setup_1contract_1version()
```

Comment:
Then you only need to check that you indeed get the unique_version in all the cases.
```suggestion
        unique_version = self.setup_1contract_1version()
```